docs/reference/how-to/size-your-shards.asciidoc

```diff
@@ -544,3 +544,50 @@ PUT _cluster/settings
   }
 }
 ----
+
+[discrete]
+==== Number of documents in the index shard cannot exceed [2147483519]
+
+NOTE: Historically, this error reported `Number of documents in the index shard cannot exceed [2147483519]` where 
+`index` was referencing a Lucene index which correlates to an <<glossary-shard,Elasticsearch shard>>. 
```

Comment:
I don't think it works well to include information about other versions in the docs. We've done it before, and it tends to languish for years longer than needed. It's especially bad to include it as a `NOTE` at the top of the relevant section, since that pushes the important content down. I suggest just removing this. If we decide to backport these docs to a version before 8.15 then we can include the older message (which NB isn't exactly what's written here).

```suggestion
```